src/app/util/CHIPDeviceCallbacksMgr.h

```diff
@@ -81,7 +83,7 @@ class DLL_EXPORT CHIPDeviceCallbacksMgr
         Callback::Cancelable * ca = &queue;
         while (ca != nullptr && ca->mNext != &queue)
         {
-            if (*reinterpret_cast<T *>(&ca->mNext->mInfoPtr) == info)
+            if (*reinterpret_cast<T *>(ca->mNext->mInfoPtr) == info)
```

Comment:
Because the queues are separated by type, this is "statically" safe by invariants, but not safe to other new queues being added that may mix callback types. If a PR comes in that mixes two types of info in one callback queue, this callback queue is unsafe.